src/Connections/GithubRepo.ts

```diff
@@ -1133,6 +1133,7 @@ ${event.release.body}`;
     public async provisionerUpdateConfig(userId: string, config: Record<string, unknown>) {
         const validatedConfig = GitHubRepoConnection.validateState(config);
         await this.as.botClient.sendStateEvent(this.roomId, GitHubRepoConnection.CanonicalEventType, this.stateKey, validatedConfig);
+        this.state = validatedConfig;
```

Comment:
Oh, I think because we sent a state event to the HS, it sends a event back to us and updates the connection for us...I don't know whether this is needed.

Reply:
It is, because after a save, the widget UI updates to match the state object returned by the "update" widget API call.  Without updating the state object here, the widget API response returns the state object from before the update, and the widget UI won't show the change you just made.

Reply:
Actually, the stale updates are partially fixed by correctly awaiting on the update, but setting the state here still prevents a flicker in the UI:

[hookshotWidgetPrefixFlicker.webm](https://user-images.githubusercontent.com/3271094/193683243-5609971a-ce81-46d8-aedc-8b2f38f0acdd.webm)

Reply:
Turns out there's more to this after all -- will return with a full fix

Reply:
ecb3796 is a quick fix for this:

[hookshotWidgetPrefixNoFlicker.webm](https://user-images.githubusercontent.com/3271094/193728786-147f1e1c-07de-4875-a945-79c84cc5f091.webm)

Updating `this.state` in each of the `provisionerUpdateConfig`s is still useful, as the current flow of saving a config & the widget picking it up is quite racy:
1. Widget makes widget API request to update the connection's config state
2. Bridge calls `provisionerUpdateConfig` to send a state event with the new config state, and responds with 200 + the config state object
3. Bridge eventually sees that config state again as a state event change from the homeserver, and updates the connection's config state JS object
4. Widget makes new widget API request to get the latest config states of all connections

Steps 3 and 4 each happen asynchronously and can occur in any order, meaning that if the config state isn't eagerly saved in Step 2, the bridge may answer the widget API request in Step 4 with a stale state before the HS sends the state event back to the bridge in Step 3.

Making matters more confusing is that the bridge's response in Step 2 includes the config state object, which 1) is a stale object if it isn't eagerly saved, and 2) isn't used by the widget's response logic anyways.

With all that said, the current change should be plenty for avoiding these problems.  I've opened #516 to track tightening up the handling of bot-sent state events, when there's time/interest to look into it.